clouddriver-google/src/main/java/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.java

```diff
@@ -0,0 +1,1391 @@
+                  sourceGroupName));
+        } else {
+          namedPorts =
+              sourceServerGroup.getNamedPorts().entrySet().stream()
+                  .map(entry -> new NamedPort().setName(entry.getKey()).setPort(entry.getValue()))
+                  .collect(Collectors.toList());
+        }
+      } else {
+        if (description.getLoadBalancingPolicy().getNamedPorts() != null) {
+          namedPorts = description.getLoadBalancingPolicy().getNamedPorts();
+        } else if (description.getLoadBalancingPolicy().getListeningPort() != null) {
+          log.warn(
+              "Deriving named ports from deprecated 'listeningPort' attribute. Please update your deploy description to use 'namedPorts'.");
+          namedPorts.add(
+              new NamedPort()
+                  .setName(GoogleHttpLoadBalancingPolicy.HTTP_DEFAULT_PORT_NAME)
+                  .setPort(description.getLoadBalancingPolicy().getListeningPort()));
+        }
+      }
+
+      if (namedPorts.isEmpty()) {
+        log.warn(
+            "Could not locate named port on either load balancing policy or source server group. Setting default named port.");
+        namedPorts.add(
+            new NamedPort()
+                .setName(GoogleHttpLoadBalancingPolicy.HTTP_DEFAULT_PORT_NAME)
+                .setPort(GoogleHttpLoadBalancingPolicy.getHTTP_DEFAULT_PORT()));
+      }
+      instanceGroupManager.setNamedPorts(namedPorts);
+    }
+  }
+
+  protected void createInstanceGroupManagerFromInput(
+      BasicGoogleDeployDescription description,
+      InstanceGroupManager instanceGroupManager,
+      LoadBalancerInfo lbInfo,
+      String serverGroupName,
+      String region,
+      Task task)
+      throws IOException {
+    if (description.getRegional()) {
+      setDistributionPolicyToInstanceGroup(description, instanceGroupManager);
+      String targetLink =
+          createRegionalInstanceGroupManagerAndWait(
+              description, lbInfo, serverGroupName, region, instanceGroupManager, task);
+      createRegionalAutoscaler(description, serverGroupName, targetLink, region, task);
+    } else {
+      String targetLink =
+          createInstanceGroupManagerAndWait(
+              description, lbInfo, serverGroupName, instanceGroupManager, task);
+      createAutoscaler(description, serverGroupName, targetLink, task);
+    }
+  }
+
+  protected void setDistributionPolicyToInstanceGroup(
+      BasicGoogleDeployDescription description, InstanceGroupManager instanceGroupManager) {
+    if (description.getDistributionPolicy() != null) {
+      DistributionPolicy distributionPolicy = new DistributionPolicy();
+
+      if (description.getSelectZones()
+          && !description.getDistributionPolicy().getZones().isEmpty()) {
+        log.info(
+            String.format(
+                "Configuring explicit zones selected for regional server group: %s",
+                description.getDistributionPolicy().getZones().get(0)));
```

Comment:
get 0 - just the one?  NOT checked old code... but we're streaming below which may mean multiple zones...

Reply:
In the old code it only logs the first element but you are right. It should log all the selected regions because they are use in the distribution policy.